carbon/app.py

```diff
@@ -98,20 +97,192 @@
+    add_child(record, "PUBLISHER", article["PUBLISHER"])
+    xml_file.write(record)
+
+
+def _add_person(xml_file: IO, person: dict[str, Any]) -> None:
+    """Create an XML element representing a person and write it to an output file.
+
+    The function will create a single 'record' element that contains subelements
+    representing fields from the 'HR_PERSON_EMPLOYEE_LIMITED', 'HR_ORG_UNIT',
+    and 'ORCID_TO_MITID' tables. The 'record' element is written to
+    to an XML file.
+
+    Args:
+        xml_file (IO): A file-like object (stream) that writes to an XML file.
+        person (dict[str, Any]): A record from from a data model that includes fields
+          from the 'HR_PERSON_EMPLOYEE_LIMITED', 'HR_ORG_UNIT', and 'ORCID_TO_MITID'
+          tables.
+    """
+    record = ET.Element("record")
+    add_child(record, "field", person["MIT_ID"], name="[Proprietary_ID]")
+    add_child(record, "field", person["KRB_NAME_UPPERCASE"], name="[Username]")
+    add_child(
+        record,
+        "field",
+        initials(person["FIRST_NAME"], person["MIDDLE_NAME"]),
+        name="[Initials]",
+    )
+    add_child(record, "field", person["LAST_NAME"], name="[LastName]")
+    add_child(record, "field", person["FIRST_NAME"], name="[FirstName]")
+    add_child(record, "field", person["EMAIL_ADDRESS"], name="[Email]")
+    add_child(record, "field", "MIT", name="[AuthenticatingAuthority]")
+    add_child(record, "field", "1", name="[IsAcademic]")
+    add_child(record, "field", "1", name="[IsCurrent]")
+    add_child(record, "field", "1", name="[LoginAllowed]")
+    add_child(
+        record,
+        "field",
+        group_name(person["DLC_NAME"], person["PERSONNEL_SUBAREA_CODE"]),
+        name="[PrimaryGroupDescriptor]",
+    )
+    add_child(
+        record,
+        "field",
+        hire_date_string(person["ORIGINAL_HIRE_DATE"], person["DATE_TO_FACULTY"]),
+        name="[ArriveDate]",
+    )
+    add_child(
+        record,
+        "field",
+        person["APPOINTMENT_END_DATE"].strftime("%Y-%m-%d"),
+        name="[LeaveDate]",
+    )
+    add_child(record, "field", person["ORCID"], name="[Generic01]")
+    add_child(record, "field", person["PERSONNEL_SUBAREA_CODE"], name="[Generic02]")
+    add_child(record, "field", person["ORG_HIER_SCHOOL_AREA_NAME"], name="[Generic03]")
+    add_child(record, "field", person["DLC_NAME"], name="[Generic04]")
+    add_child(record, "field", person.get("HR_ORG_LEVEL5_NAME"), name="[Generic05]")
+    xml_file.write(record)
+
+
+def add_child(
+    parent: ET._Element,  # noqa: SLF001
+    element: str,
+    text: str | None = None,
+    **kwargs: str,
+) -> ET._Element:  # noqa: SLF001
+    """Add a subelement to an existing element.
+
+    Args:
+        parent (ET._Element): The root element.
+        element: The tag name assigned to the subelement.
+        text (str | None, optional): _description_. Defaults to None.
+        **kwargs (str): See lxml.etree.SubElement for more details.
```

Comment:
Since it looks only `name=` is being used as in `**kwargs`, perhaps just detail what's happening there?